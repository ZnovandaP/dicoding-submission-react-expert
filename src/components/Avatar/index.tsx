import * as React from 'react';
import Image from 'next/image';
import cn from '@/utils/cn';

type AvatarProps = {
  src: string;
  alt: string;
  size?: 'sm' | 'md' | 'lg' | 'xl';
  className?: string;
};

export default function Avatar({
  size = 'sm', alt, src, className,
}: AvatarProps) {
  return (
    <Image
      src={src}
      alt={alt}
      width={500}
      height={500}
      loading="lazy"
      className={cn(
        'rounded-full',

        size === 'sm' && 'h-8 w-8',

        size === 'md' && 'h-12 w-12',

        size === 'md' && 'h-16 w-16',

        size === 'xl' && 'h-20 w-20',

        className ?? '',
      )}
    />
  );
}
