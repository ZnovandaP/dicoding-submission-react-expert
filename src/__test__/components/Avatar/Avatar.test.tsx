/* eslint-disable react/react-in-jsx-scope */

/*
Skenario test component Avatar:
* should render correctly component Avatar
* should render component Avatar with size 'sm'
* should render component Avatar with size 'md'
* should render component Avatar with size 'lg'
* should render component Avatar with size 'xl'
*/

import Avatar from '@/components/Avatar';
import {
  describe, it, expect,
} from '@jest/globals';
import { render, screen } from '@testing-library/react';

describe('test Avatar component', () => {
  it('should render correctly component Avatar', () => {
    render(<Avatar src="/user.png" alt="user avatar" />);

    const imageAvatar = screen.getByRole('img');

    expect(imageAvatar).toMatchSnapshot();
  });

  it('should render component Avatar with size sm', () => {
    render(<Avatar src="/user.png" alt="user avatar" size="sm" />);

    const imageAvatar = screen.getByRole('img');

    expect(imageAvatar.classList.contains('w-8')).toBeTruthy();
  });

  it('should render component Avatar with size md', () => {
    render(<Avatar src="/user.png" alt="user avatar" size="md" />);

    const imageAvatar = screen.getByRole('img');

    expect(imageAvatar.classList.contains('w-12')).toBeTruthy();
  });

  it('should render component Avatar with size lg', () => {
    render(<Avatar src="/user.png" alt="user avatar" size="lg" />);

    const imageAvatar = screen.getByRole('img');

    expect(imageAvatar.classList.contains('w-16')).toBeTruthy();
  });

  it('should render component Avatar with size xl', () => {
    render(<Avatar src="/user.png" alt="user avatar" size="xl" />);

    const imageAvatar = screen.getByRole('img');

    expect(imageAvatar.classList.contains('w-20')).toBeTruthy();
  });
});
