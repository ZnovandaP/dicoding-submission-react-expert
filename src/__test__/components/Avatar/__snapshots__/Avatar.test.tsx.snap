// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Avatar component should render correctly component Avatar 1`] = `
<img
  alt="user avatar"
  class="rounded-full h-8 w-8"
  data-nimg="1"
  decoding="async"
  height="500"
  loading="lazy"
  src="/_next/image?url=%2Fuser.png&w=1080&q=75"
  srcset="/_next/image?url=%2Fuser.png&w=640&q=75 1x, /_next/image?url=%2Fuser.png&w=1080&q=75 2x"
  style="color: transparent;"
  width="500"
/>
`;
